src/base/settingsstorage.cpp

```diff
@@ -56,8 +56,8 @@ namespace
         MappingTable(std::initializer_list<std::pair<QString, QString>> list)
         {
             reserve(static_cast<int>(list.size()));
-            for (std::initializer_list<std::pair<QString, QString>>::const_iterator it = list.begin(); it != list.end(); ++it)
-                insert(it->first, it->second);
+            for (const auto &i : list)
```

Comment:
I would still not recommend to use range-based `for` with Qt containers.
From Qt doc:
>An alternative to Qt's `foreach` loop is the range-based `for` that is part of C++ 11 and newer. However, keep in mind that the range-based `for` might force a Qt container to detach, whereas `foreach` would not. [...] If in doubt, prefer `foreach` for Qt containers, and range based `for` for STL ones.

Besides, I just used the same code that is used in Qt5 QHash constructor.